### DISCUSSION TITLE
How do I add two S3 backend to two different environments using terragrunt.hcl?

### DISCUSSION THREAD

**Discussion by shreyas-wego** [gruntwork-io/knowledge-base#668] [How do I]
Hi Team,

We are having two different environments Staging and Production and we are maintaining `staging.tfvars` and `production.tfvars` respectively. We want to store our state files in S3 backend into two different buckets in Staging and Production accounts respectively.

Is it possible to just have 1 `terragrunt.hcl` file to add two remote_state configurations? Any suggestions would be helpful.

---

<ins datetime="2023-02-21T04:53:40Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109916">Tracked in ticket #109916</a></p>
</ins>

**Comment by ellisonc** [comment]
Take a look at the [example repo](https://github.com/gruntwork-io/terragrunt-infrastructure-live-example).  In the top level `terragrunt.hcl` file we are building the bucket name and key using local variables from `account.hcl`, `region.hcl`, and `env.hcl`.  You could use this same pattern to set the bucket name on a per-environment basis.  Your terragrunt file might look something like: 

```
locals {
  ...
  # Automatically load environment-level variables
  environment_vars = read_terragrunt_config(find_in_parent_folders("env.hcl"))

  # Extract the variables we need for easy access
  ...
  env = local.environment_vars.locals.environment
}
...
# Configure Terragrunt to automatically store tfstate files in an S3 bucket
remote_state {
  backend = "s3"
  config = {
    ...
    bucket = "${local.env}-terragrunt-example-terraform-state-${local.account_name}-${local.aws_region}"
    ...
  }
  ...
}
...
```

**Reply by shreyas-wego** [reply]
Hi @ellisonc thank you for the response. I have one more question here, is there a way to dynamically pass the AWS profile name associated to a particular account?

**Reply by ellisonc** [reply]
@shreyas-wego Yes, if you take a look at the same repo from earlier the sub-folders contain account.hcl files.  At gruntwork we use these files for account specific variables like aws_profile.  In the top level `terragrunt.hcl` that variable would be available as `local.account_vars.locals.aws_profile`

**Reply by shreyas-wego** [reply]
Thank you @ellisonc although I didn't `local.account_vars.locals.aws_profile` in top level `terragrunt.hcl` file but I understood the flow of how things are working. I will definitely try this out and let you know on this thread on how it goes. Thanks again! Have a good day!
